PPSSPPHeadless fails at linking on riscv64 platform

### Platform

Linux / BSD

### Compiler and build tool versions

Clang 14.0.6 + CMake 3.24.0

### Operating system version

Arch Linux RISC-V port

### Build commands used

```
 cmake -S ppsspp -B build-sdl -G Ninja \
    -DCMAKE_BUILD_TYPE=None \
    -DCMAKE_SKIP_RPATH=ON \
    -DHEADLESS=ON \
    -DOpenGL_GL_PREFERENCE=GLVND \
    -DUSE_SYSTEM_LIBZIP=ON \
    -DUSE_SYSTEM_SNAPPY=ON \
    -DUSE_SYSTEM_ZSTD=ON \
    -DUSING_QT_UI=OFF
  cmake --build build-sdl
```

### What happens

```
[582/583] Linking CXX executable PPSSPPHeadless
FAILED: PPSSPPHeadless
: && /usr/bin/clang++   CMakeFiles/PPSSPPHeadless.dir/headless/Headless.cpp.o CMakeFiles/PPSSPPHeadless.dir/headless/StubHost.cpp.o CMakeFiles/PPSSPPHeadless.dir/headless/Compare.cpp.o CMakeFiles/PPSSPPHeadless.dir/headless/SDLHeadlessHost.cpp.o -o PPSSPPHeadless  lib/libCore.a  /usr/lib/libzstd.so  lib/libnative.a  lib/libCommon.a  /lib/libsnappy.so  /usr/lib/libzip.so  /usr/lib/libpng.so  lib/libvma.a  lib/libgason.a  lib/libudis86.a  -lrt  /usr/lib/libSDL2.so  /usr/lib/libGLEW.so  lib/libkirk.a  lib/libcityhash.a  lib/libsfmt19937.a  lib/libxbrz.a  lib/libxxhash.a  lib/libglslang.a  lib/libSPIRV.a  lib/libMachineIndependent.a  lib/libOGLCompiler.a  lib/libOSDependent.a  lib/libGenericCodeGen.a  lib/libspirv-cross-glsl.a  lib/libspirv-cross-core.a  lib/libarmips.a  /usr/lib/libOpenGL.so  -ldl  /root/ppsspp/ffmpeg/linux/riscv64/lib/libavformat.a  /root/ppsspp/ffmpeg/linux/riscv64/lib/libavcodec.a  /root/ppsspp/ffmpeg/linux/riscv64/lib/libswresample.a  /root/ppsspp/ffmpeg/linux/riscv64/lib/libswscale.a  /root/ppsspp/ffmpeg/linux/riscv64/lib/libavutil.a  /usr/lib/libz.so  lib/libdiscord-rpc.a  lib/libminiupnpc.a && :
/usr/bin/ld: lib/libCommon.a(FakeJit.cpp.o): in function `no symbol':
FakeJit.cpp:(.text+0x88): undefined reference to `JitBlockCache::JitBlockCache(MIPSState*, CodeBlockCommon*)'
/usr/bin/ld: FakeJit.cpp:(.text+0xd2): undefined reference to `JitBlockCache::Init()'
/usr/bin/ld: FakeJit.cpp:(.text+0x100): undefined reference to `JitBlockCache::~JitBlockCache()'
/usr/bin/ld: FakeJit.cpp:(.text+0x448): undefined reference to `JitBlockCache::Clear()'
/usr/bin/ld: FakeJit.cpp:(.text+0x4ac): undefined reference to `JitBlockCache::InvalidateICache(unsigned int, unsigned int)'
/usr/bin/ld: lib/libCommon.a(FakeJit.cpp.o): in function `.LBB51_1':
FakeJit.cpp:(.text._ZN8MIPSComp7FakeJitD2Ev[_ZN8MIPSComp7FakeJitD2Ev]+0x48): undefined reference to `JitBlockCache::~JitBlockCache()'
/usr/bin/ld: lib/libCommon.a(FakeJit.cpp.o): in function `no symbol':
FakeJit.cpp:(.text._ZN8MIPSComp7FakeJit22SaveAndClearEmuHackOpsEv[_ZN8MIPSComp7FakeJit22SaveAndClearEmuHackOpsEv]+0x18): undefined reference to `JitBlockCache::SaveAndClearEmuHackOps()'
/usr/bin/ld: FakeJit.cpp:(.text._ZN8MIPSComp7FakeJit22RestoreSavedEmuHackOpsESt6vectorIjSaIjEE[_ZN8MIPSComp7FakeJit22RestoreSavedEmuHackOpsESt6vectorIjSaIjEE]+0x3e): undefined reference to `JitBlockCache::RestoreSavedEmuHackOps(std::vector<unsigned int, std::allocator<unsigned int> >)'
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
[583/583] Linking CXX executable PPSSPPSDL
```
Note that PPSSPPSDL is able to link successfully but PPSSPPHeadless isn't.

### PPSSPP version affected

v1.13.1-355-g0749f1465

### Last working version

_No response_

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] Make sure to run `git submodule update --init --recursive` before building.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try deleting the build directory and running the build again.
- [X] Check GitHub Actions, which builds every merge and PR.
- [X] Include logs and help us reproduce.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,6 +426,15 @@ if(WIN32)
 endif()
 
 
+set(CommonJIT
+	Core/MIPS/JitCommon/JitCommon.cpp
+	Core/MIPS/JitCommon/JitCommon.h
+	Core/MIPS/JitCommon/JitBlockCache.cpp
+	Core/MIPS/JitCommon/JitBlockCache.h
+	Core/MIPS/JitCommon/JitState.cpp
+	Core/MIPS/JitCommon/JitState.h
+)
+
 set(CommonX86
 	Common/ABI.cpp
 	Common/ABI.h
@@ -464,6 +473,7 @@ set(CommonMIPS
 source_group(MIPS FILES ${CommonMIPS})
 
 set(CommonRISCV64
+	${CommonJIT}
 	Common/RiscVCPUDetect.cpp
 	Core/MIPS/fake/FakeJit.cpp
 	Core/MIPS/fake/FakeJit.h
@@ -1626,6 +1636,7 @@ set(GPU_SOURCES
 # SHARED on ANDROID, STATIC everywhere else
 add_library(${CoreLibName} ${CoreLinkType}
 	${CoreExtra}
+	${CommonJIT}
 	Core/Config.cpp
 	Core/Config.h
 	Core/ConfigValues.h
@@ -1933,12 +1944,6 @@ add_library(${CoreLibName} ${CoreLinkType}
 	Core/FileLoaders/RamCachingFileLoader.h
 	Core/FileLoaders/RetryingFileLoader.cpp
 	Core/FileLoaders/RetryingFileLoader.h
-	Core/MIPS/JitCommon/JitCommon.cpp
-	Core/MIPS/JitCommon/JitCommon.h
-	Core/MIPS/JitCommon/JitBlockCache.cpp
-	Core/MIPS/JitCommon/JitBlockCache.h
-	Core/MIPS/JitCommon/JitState.cpp
-	Core/MIPS/JitCommon/JitState.h
 	Core/MIPS/MIPS.cpp
 	Core/MIPS/MIPS.h
 	Core/MIPS/MIPSAnalyst.cpp

